cpp/dolfinx/fem/interpolate.h

```diff
@@ -107,8 +147,495 @@ void interpolate_from_any(Function<T>& u, const Function<T>& v)
+      {
+        sendingOffsets[i] += allPointsToSend[nProcs * j + i];
+      }
+    }
+
+    xt::xtensor<double, 2> pointsToReceive(
+        xt::shape(
+            {nPointsToReceive / 3, static_cast<decltype(nPointsToReceive)>(3)}),
+        0);
+    MPI_Win window;
+    MPI_Win_create(pointsToReceive.data(), sizeof(double) * nPointsToReceive,
+                   sizeof(double), MPI_INFO_NULL, MPI_COMM_WORLD, &window);
+    MPI_Win_fence(0, window);
+
+    for (int i = 0; i < nProcs; ++i)
+    {
+      MPI_Put(pointsToSend[i].data(), pointsToSend[i].size(), MPI_DOUBLE, i,
+              sendingOffsets[i], pointsToSend[i].size(), MPI_DOUBLE, window);
+    }
+
+    MPI_Win_fence(0, window);
+    MPI_Win_free(&window);
+
+    // Each process will now check at which points it can evaluate
+    // the interpolating function, and note that down in evaluationCells
+    std::vector<std::int32_t> evaluationCells(pointsToReceive.shape(0), -1);
+
+    const auto connectivity = v.function_space()->mesh()->geometry().dofmap();
+
+    // This BBT is useful for fast lookup of which cell contains a given point
+    dolfinx::geometry::BoundingBoxTree bbt(*v.function_space()->mesh(), tdim,
+                                           0.0001);
+
+    const auto xv = v.function_space()->mesh()->geometry().x();
+
+    if (tdim == 3)
+    {
+      // For each point at which the source function needs to be evaluated
+      for (decltype(pointsToReceive.shape(0)) i = 0;
+           i < pointsToReceive.shape(0); ++i)
+      {
+        // Get its coordinates
+        const double xp = pointsToReceive(i, 0);
+        const double yp = pointsToReceive(i, 1);
+        const double zp = pointsToReceive(i, 2);
+
+        // For each cell that might contain that point
+        for (const auto& j :
+             dolfinx::geometry::compute_collisions(bbt, {xp, yp, zp}))
+        {
+          // Get the vertexes that belong to that cell
+          const auto vtx = connectivity.links(j);
+
+          const auto x1 = xv[3 * vtx[0]];
+          const auto y1 = xv[3 * vtx[0] + 1];
+          const auto z1 = xv[3 * vtx[0] + 2];
+          const auto x2 = xv[3 * vtx[1]];
+          const auto y2 = xv[3 * vtx[1] + 1];
+          const auto z2 = xv[3 * vtx[1] + 2];
+          const auto x3 = xv[3 * vtx[2]];
+          const auto y3 = xv[3 * vtx[2] + 1];
+          const auto z3 = xv[3 * vtx[2] + 2];
+          const auto x4 = xv[3 * vtx[3]];
+          const auto y4 = xv[3 * vtx[3] + 1];
+          const auto z4 = xv[3 * vtx[3] + 2];
+
+          // We compute the barycentric coordinates of the given point
+          // in the cell at hand
+          const xt::xarray<double> A = {{x1 - x4, x2 - x4, x3 - x4},
+                                        {y1 - y4, y2 - y4, y3 - y4},
+                                        {z1 - z4, z2 - z4, z3 - z4}};
+          const xt::xarray<double> b = {xp - x4, yp - y4, zp - z4};
+
+          const xt::xarray<double> l = xt::linalg::solve(A, b);
```

Comment:
Note that xt::linalg was removed from dolfinx in https://github.com/FEniCS/dolfinx/pull/1703

Reply:
I request it explicitly in the `CMakeLists.txt` file, but I agree this should be addressed better.

Reply:
That piece of code was removed.